distributed/shuffle/_worker_plugin.py

```diff
@@ -144,6 +156,29 @@ async def get_or_create(self, spec: ShuffleSpec, key: str) -> ShuffleRun:
             raise shuffle_run._exception
         return shuffle_run
 
+    async def _fetch(
+        self,
+        shuffle_id: ShuffleId,
+        spec: ShuffleSpec | None = None,
+        key: str | None = None,
+    ) -> ShuffleRunSpec:
+        # FIXME: This should never be ToPickle[ShuffleRunSpec]
+        result: ShuffleRunSpec | ToPickle[ShuffleRunSpec]
+        if spec is None:
+            result = await self._plugin.worker.scheduler.shuffle_get(
+                id=shuffle_id,
+                worker=self._plugin.worker.address,
+            )
+        else:
+            result = await self._plugin.worker.scheduler.shuffle_get_or_create(
+                spec=ToPickle(spec),
+                key=key,
+                worker=self._plugin.worker.address,
+            )
+        if isinstance(result, ToPickle):
```

Comment:
out of scope: I'm not sure I understand why this may or may not be wrapped in ToPickle?

Reply:
It's something in the comms, haven't dug too deep into it.